core/store/src/adapter/trie_store.rs

```diff
@@ -188,9 +169,18 @@ impl<'a> TrieStoreUpdateAdapter<'a> {
     }
 }
 
-pub fn get_key_from_shard_uid_and_hash(shard_uid: ShardUId, hash: &CryptoHash) -> [u8; 40] {
+/// Reads shard_uid mapping for given shard.
+/// If the mapping does not exist, it means that `shard_uid` maps to itself
+pub fn read_shard_uid_mapping_from_db(store: &Store, shard_uid: ShardUId) -> MappedShardUId {
```

Comment:
Can you add a comment explaining the perf considerations regarding reading the mapped shard uid from db every time? Something like: 
```
Please note that the mapped shard uid is read from db each time which may seem slow. In practice the `StateShardUIdMapping` is very small and should always be stored in the RocksDB cache. The deserialization of ShardUId is also very cheap.
```